kolibri/plugins/management/assets/src/device_management/views/manage-content-page/task-progress.vue

```diff
@@ -0,0 +1,195 @@
+        color="primary"
+      />
+    </div>
+
+    <div class="progress-messages dtc">
+      <span class="percentage">{{ progressMessage }}</span>
+    </div>
+
+    <div class="buttons dtc">
+      <k-button
+        :text="taskHasCompleted ? $tr('close') : $tr('cancel')"
+        :primary="false"
+        :raised="false"
+        @click="endTask()"
+      />
+    </div>
+
+  </div>
+
+</template>
+
+
+<script>
+
+  import UiProgressLinear from 'keen-ui/src/UiProgressLinear';
+  import kButton from 'kolibri.coreVue.components.kButton';
+  import round from 'lodash/round';
+  import { refreshChannelList } from '../../state/actions/manageContentActions';
+  import { cancelTask } from '../../state/actions/taskActions';
+  import { TaskTypes, TaskStatuses } from '../../constants';
+
+  const RequiredString = {
+    type: String,
+    required: true,
+  };
+
+  export default {
+    name: 'taskProgress',
+    components: {
+      UiProgressLinear,
+      kButton,
+    },
+    props: {
+      type: RequiredString,
+      status: RequiredString,
+      percentage: {
+        type: Number,
+        required: true,
+      },
+      id: RequiredString,
+    },
+    computed: {
+      TaskStatuses: () => TaskStatuses,
+      stageText() {
+        if (this.status === TaskStatuses.RUNNING) {
+          switch (this.type) {
+            case TaskTypes.REMOTE_IMPORT:
+            case TaskTypes.LOCAL_IMPORT:
+              return this.$tr('importingContent');
+            case TaskTypes.LOCAL_EXPORT:
+              return this.$tr('exportingContent');
+            default:
+              return '';
+          }
+        }
+        if (this.taskHasFailed) {
+          return this.$tr('taskHasFailed');
+        }
+        if (this.taskHasCompleted) {
+          return this.$tr('finished');
+        }
+        if (this.taskIsPreparing) {
+          return this.$tr('preparingTask');
+        }
+      },
+      taskHasFailed() {
+        return this.status === TaskStatuses.FAILED;
+      },
+      taskHasCompleted() {
+        return this.status === TaskStatuses.COMPLETED;
+      },
+      taskIsPreparing() {
+        return this.status === TaskStatuses.QUEUED || this.status === TaskStatuses.SCHEDULED;
+      },
+      formattedPercentage() {
+        return round(this.percentage * 100, 2);
+      },
+      progressMessage() {
+        if (this.percentage > 0) {
+          return this.formattedPercentage + '%';
+        }
+        return '';
+      },
+      progressBarType() {
+        return this.taskIsPreparing ? 'indeterminate' : 'determinate';
+      },
+    },
+    methods: {
+      endTask() {
+        if (this.taskHasCompleted) {
+          this.$emit('taskcomplete');
+          this.refreshChannelList();
+        }
+        this.cancelTask(this.id);
+      },
+    },
+    vuex: {
+      actions: {
+        cancelTask,
+        refreshChannelList,
+      },
+    },
+    $trs: {
+      importingContent: 'Importing content\u2026',
```

Comment:
I wonder if having `...` instead of the ellipsis unicode, is better for translations.

Reply:
Should be able to use the real unicode character `…`

Reply:
I copied this from the old 'task-status.vue'. I replaced it with the plain '…' character (different than '...').